monai/losses/focal_loss.py

```diff
@@ -9,16 +9,18 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Optional, Union
+import warnings
+from typing import Optional, Sequence, Union
 
 import torch
 import torch.nn.functional as F
-from torch.nn.modules.loss import _WeightedLoss
+from torch.nn.modules.loss import _Loss
 
+from monai.networks import one_hot
 from monai.utils import LossReduction
 
 
-class FocalLoss(_WeightedLoss):
+class FocalLoss(_Loss):
```

Comment:
why changing this base class? I think the previous one has better support of torchscript

Reply:
In some cases, like a 2 class classification task, a common way is to input a single `alpha` (as the paper shown [here](https://arxiv.org/pdf/1708.02002.pdf)). That's why I change the type to `Optional[Union[Sequence[float], float, torch.Tensor]]`. As for `_WeightedLoss `, the type of `weight` can only be `torch.Tensor`, thus I changed it into `_Loss`. What do you think?